lightnode/kvstore/store.go

```diff
@@ -0,0 +1,41 @@
+package kvstore
+
+import (
+	"errors"
+	"github.com/syndtr/goleveldb/leveldb/iterator"
+)
+
+// Store implements a key-value store. May be backed by a database like LevelDB.
+type Store interface {
+
+	// Put stores the given key / value pair in the database, overwriting any existing value for that key.
+	Put(key []byte, value []byte) error
+
+	// Get retrieves the value for the given key. Returns a ErrNotFound error if the key does not exist.
+	Get(key []byte) ([]byte, error)
+
+	// Delete removes the key from the database. Does not return an error if the key does not exist.
+	Delete(key []byte) error
+
+	// DeleteBatch atomically removes a list of keys from the database.
+	DeleteBatch(keys [][]byte) error
+
+	// WriteBatch atomically writes a list of key / value pairs to the database. The key at index i in the keys slice
+	// corresponds to the value at index i in the values slice.
+	WriteBatch(keys, values [][]byte) error
+
```

Comment:
What if the client wants to both read and write kvs atomically?

Reply:
That's a legitimate feature for a general purpose database, but I don't think it will be necessary for the features we are planning for light nodes. I intentionally avoided adding features that won't be needed for light nodes.